cpp/src/plasma/common.cc

```diff
@@ -29,6 +30,13 @@ namespace plasma {
 
 using arrow::Status;
 
+std::mt19937 RandomlySeededMersenneTwister() {
+  auto seed = static_cast<std::mt19937::result_type>(
```

Comment:
I don't think you need to compute your own seed, just use `std::random_device` as shown in the example here: https://en.cppreference.com/w/cpp/numeric/random

Reply:
Note that C++ doesn't allow use to correctly seed the MT engine, but for this purpose it probably doesn't matter much. See http://www.pcg-random.org/posts/cpp-seeding-surprises.html

Reply:
Thanks for the feedback! Unfortunately std::random_device breaks the valgrind that ships with ubuntu 16.04 (it uses a processor instruction that is not implemented in that version of valgrind, but if you compile the latest valgrind it works).

So our choices are:
- Read randomness from /dev/urandom (might not be portable)
- Come up with some good way to seed ourselves (most likely doesn't give the best possible randomness)
- Use random_device, upgrade valgrind on Travis and document somewhere that a newer version of valgrind is needed

Let me know if you have thoughts on this!

Reply:
Ah, right, I got this locally too. Then keep the current seed and add a comment about Valgrind?